### PR TITLE
Add preflight check for standalone pods

### PR DESCRIPTION
**Pull request by pprach**

### CODE REVIEW

File: ansible/playbooks/roles/preflight/tasks/upgrade.yml [hunk 1]
```diff
@@ -15,6 +15,20 @@
...
+
+    - name: Check number of unmanaged pods
+      become: true
+      command: kubectl get pods --all-namespaces -o custom-columns=CONTROLLER:.metadata.ownerReferences[].kind
+      register: no_unmanaged_pods
+      changed_when: false
+
+    - name: Check if cluster has any unmanaged pods
+      assert:
+        that: "{{ no_unmanaged_pods.stdout_lines | regex_findall('<none>') | length }} < 1"
```
Comment (atsikham) [inline ansible/playbooks/roles/preflight/tasks/upgrade.yml]: May it be negative? 😄

**Review by atsikham — CHANGES_REQUESTED** [review]

**Comment by atsikham** [comment]
Please update changelog.

File: ansible/playbooks/roles/preflight/tasks/upgrade.yml (hunk 1 above)
Comment (seriva) [reply ansible/playbooks/roles/preflight/tasks/upgrade.yml]: @atsikham Hooray for defensive programming!

**Comment by seriva** [comment]
@pprach Pls address comments so we can get this merged.

**Review by seriva — DISMISSED** [review]

File: ansible/playbooks/roles/preflight/tasks/upgrade.yml [hunk 2]
```diff
@@ -15,6 +15,20 @@
...
+
+    - name: Check number of unmanaged pods
+      become: true
+      command: kubectl get pods --all-namespaces -o custom-columns=CONTROLLER:.metadata.ownerReferences[].kind
+      register: no_unmanaged_pods
+      changed_when: false
+
+    - name: Check if cluster has any unmanaged pods
+      assert:
+        that: "{{ no_unmanaged_pods.stdout_lines | regex_findall('<none>') | length }} != 0"
```
Comment (przemyslavic) [inline ansible/playbooks/roles/preflight/tasks/upgrade.yml]: shouldn't it be == 0?

Comment (atsikham) [reply ansible/playbooks/roles/preflight/tasks/upgrade.yml]: Waiting for reply

Comment (pprach) [reply ansible/playbooks/roles/preflight/tasks/upgrade.yml]: It should be my fault. Fixed now.

**Review by seriva — APPROVED** [review]

**Review by atsikham — APPROVED** [review]

**Review by przemyslavic — APPROVED** [review]

**Comment by przemyslavic** [comment]
Cluster with unmanaged pods:
```
2021-12-10T09:52:45.3874626Z [38;21m09:52:45 INFO cli.engine.ansible.AnsibleCommand - TASK [preflight : Check number of unmanaged pods] ******************************[0m
2021-12-10T09:52:46.1654855Z [38;21m09:52:46 INFO cli.engine.ansible.AnsibleCommand - ok: [ec2-1-1-1-1.eu-west-1.compute.amazonaws.com -> ec2-1-1-1-1.eu-west-1.compute.amazonaws.com][0m
2021-12-10T09:52:46.1915746Z [38;21m09:52:46 INFO cli.engine.ansible.AnsibleCommand - [0m
2021-12-10T09:52:46.1917785Z [38;21m09:52:46 INFO cli.engine.ansible.AnsibleCommand - TASK [preflight : Check if cluster has any unmanaged pods] *********************[0m
2021-12-10T09:52:46.2347373Z [31;21m09:52:46 ERROR cli.engine.ansible.AnsibleCommand - fatal: [ec2-1-1-1-1.eu-west-1.compute.amazonaws.com -> ec2-1-1-1-1.eu-west-1.compute.amazonaws.com]: FAILED! => {"assertion": "2 == 0", "changed": false, "evaluated_to": false, "msg": "You have some pods that are not created by controller. Please remove them and then start upgrade once again."}[0m
```
After removing: ✔️
```
2021-12-10T10:05:57.4065859Z [38;21m10:05:57 INFO cli.engine.ansible.AnsibleCommand - TASK [preflight : Check number of unmanaged pods] ******************************[0m
2021-12-10T10:05:58.3668308Z [38;21m10:05:58 INFO cli.engine.ansible.AnsibleCommand - ok: [ec2-1-1-1-1.eu-west-1.compute.amazonaws.com -> ec2-1-1-1-1.eu-west-1.compute.amazonaws.com][0m
2021-12-10T10:05:58.3929472Z [38;21m10:05:58 INFO cli.engine.ansible.AnsibleCommand - [0m
2021-12-10T10:05:58.3931116Z [38;21m10:05:58 INFO cli.engine.ansible.AnsibleCommand - TASK [preflight : Check if cluster has any unmanaged pods] *********************[0m
2021-12-10T10:05:58.4338242Z [38;21m10:05:58 INFO cli.engine.ansible.AnsibleCommand - ok: [ec2-1-1-1-1.eu-west-1.compute.amazonaws.com -> ec2-1-1-1-1.eu-west-1.compute.amazonaws.com][0m
```

**Review by erzetpe — APPROVED** [review]

**Comment by przemyslavic** [comment]
:heavy_check_mark: The preflight check has been added to verify if there are any pods not created by the controller.

**Pull request merged**